### PR TITLE
Initial stub implementation for DIST-S1 PGE

### PR DESCRIPTION
**Pull request by RKuttruff**
## Description
- Implemented PGE classes (pre, post, exec) as child classes of the base PGE classes
- Added unit test stub and config YAML
- Added dockerfile + build & test scripts
- Added runconfig schema file + black files for future templates (MPC, ISO XML)

## Affected Issues
- #548 

## Testing
- Added unit test module for DIST-S1 PGE
- Added CI script for unit testing
- DID NOT add CI script for int testing nor supporting QA/comparison scripts for int testing
- DID NOT update Jenkins pipeline definitions to support this PGE in default builds (I made the changes but held back on committing them)

### CODE REVIEW

File: .ci/docker/Dockerfile_dist_s1 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# we don't run into issues when running as an outside user (via docker run -u)
+RUN set -ex \
+    && cd ${PGE_DEST_DIR} \
+    && mkdir -p ${CONDA_ROOT}/bin \
+    && cp ${PGE_DEST_DIR}/opera/scripts/*_entrypoint.sh ${CONDA_ROOT}/bin \
+    && chmod +x ${CONDA_ROOT}/bin/*_entrypoint.sh \
+    && apt-get update \
+    && apt-get install --no-install-recommends -y g++ \
+    && apt-get clean \
+    && python -m pip install -r ${PGE_DEST_DIR}/opera/requirements.txt \
```
Comment (RKuttruff) [inline .ci/docker/Dockerfile_dist_s1:60]: The pip install is failing without g++ installed
We should coordinate with ADT to have a numpy image <2 installed in the SAS so we won't have to worry about rebuilding it here, but it seems the issue is also present in the mgrs install.

**Review by collinss-jpl — CHANGES_REQUESTED** [review]
Overall looks great, mostly just some cut/paste gremlins to take care of

File: .ci/scripts/dswx_ni/build_dswx_ni.sh [hunk 2]
```diff
@@ -53,7 +53,7 @@ copy_pge_files $WORKSPACE $STAGING_DIR $PGE_NAME
 mkdir -p ${STAGING_DIR}/opera/pge/dswx_s1; cp -r ${WORKSPACE}/src/opera/pge/dswx_s1/dswx_s1_pge.py ${STAGING_DIR}/opera/pge/dswx_s1/
 
 # Create a VERSION file in the staging area to track version and build time
-printf "pge_version: ${TAG}\npge_build_datetime: ${BUILD_DATE_TIME}\n" \
+printf "pge_version: ${TAG}\pge_build_datetime: ${BUILD_DATE_TIME}\n" \
```
[the remark this answers is not in the document]
Comment (RKuttruff) [reply .ci/scripts/dswx_ni/build_dswx_ni.sh:56]: Oops, mistook that for a typo without actually reading the text of the print statement...

**Comment by RKuttruff** [comment]
@collinss-jpl Addressed findings

**Pull request merged**